### PR TITLE
[Merged by Bors] - TY 2483 Fix domain names being treated as valid URLs

### PR DESCRIPTION
**Pull request by x3ro**
In the Newscatcher API, the field `clean_url` actually contains only a domain name. This led to some confusion regarding how to treat this value, and we were trying to parse it as a URL in `Document`. We're thus renaming it to make clearer what this field actually contains and prevent future confusion.

### CODE REVIEW

File: discovery_engine_core/bindings/src/types/document/news_resource.rs [hunk 1]
```diff
@@ -59,8 +59,10 @@ pub unsafe extern "C" fn news_resource_place_of_url(place: *mut NewsResource) ->
 /// The pointer must point to a valid [`NewsResource`] memory object,
 /// it might be uninitialized.
 #[no_mangle]
-pub unsafe extern "C" fn news_resource_place_of_source_url(place: *mut NewsResource) -> *mut Url {
-    unsafe { addr_of_mut!((*place).source_url) }
+pub unsafe extern "C" fn news_resource_place_of_source_url(
```
Comment (acrrd) [inline discovery_engine_core/bindings/src/types/document/news_resource.rs:62]:
```suggestion
pub unsafe extern "C" fn news_resource_place_of_source_domain(
```

Comment (acrrd) [reply discovery_engine_core/bindings/src/types/document/news_resource.rs:62]: And the same changes also to dart.

**Comment by acrrd** [comment]
I added the same changes in dart. 
Changed some `domain.com` to `example.com`.

**Review by Robert-Steiner — APPROVED** [review]

File: discovery_engine/test/discovery_engine/utils/utils.dart [hunk 2]
```diff
@@ -77,8 +77,8 @@ void withSuccessResponse(Object initialMessage) =>
 
 final mockConfig = Configuration(
   apiKey: '**********',
-  apiBaseUrl: 'https://example-api.dev',
-  assetsUrl: 'https://ai-assets.dev',
+  apiBaseUrl: 'https://api.example.com',
```
[the remark this answers is not in the document]
Comment (x3ro) [reply discovery_engine/test/discovery_engine/utils/utils.dart:80]: `example.com` is fine afaik. See [here](http://example.com)

**Comment by x3ro** [comment]
bors merge

File: discovery_engine/test/discovery_engine/utils/utils.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (acrrd) [reply discovery_engine/test/discovery_engine/utils/utils.dart:80]: `example.com` is reserved for documentation purpose: http://www.iana.org/domains/reserved

**Comment by acrrd** [comment]
bors merge

**Comment by x3ro** [comment]
🎉

**Pull request closed without merging**